### PR TITLE
Use proper tests assertions

### PR DESCRIPTION
**Pull request by lucasmichot**
Always use the proper tests assertions

### CODE REVIEW

File: tests/CarbonInterval/ToDateIntervalTest.php [hunk 1]
```diff
@@ -15,7 +15,7 @@ public function testConvertToDateInterval()
     {
         $interval = CarbonInterval::days(5)->hours(3)->minutes(50)->microseconds(123456)->invert()->toDateInterval();
 
-        $this->assertSame('DateInterval', get_class($interval));
+        $this->assertInstanceOf(DateInterval::class, $interval);
```
Comment (kylekatarnls) [inline tests/CarbonInterval/ToDateIntervalTest.php:18]: Please revert this and the similar, it's not heuristic.

```php
'DateInterval' === get_class(new CarbonInterval()) // false
new CarbonInterval() instanceof DateInterval // true
```

**Review by kylekatarnls — CHANGES_REQUESTED** [review]
Please revert non-heuristic chnges

count/array/equals checks are fine.

File: tests/CarbonInterval/ToDateIntervalTest.php (hunk 1 above)
Comment (lucasmichot) [reply tests/CarbonInterval/ToDateIntervalTest.php:18]: Thanks for the look  @kylekatarnls 
I feel the previous assertion was actually a bit misleading and would love to enforce a bit more meaning
What do you think of `$this->assertTrue(is_a($interval, DateInterval::class));` ?

Comment (kylekatarnls) [reply tests/CarbonInterval/ToDateIntervalTest.php:18]: `is_a` has the same problem, it also makes the check pointless, where checking that we got a raw `DateInterval` and we don't want any kind of sub-class of `DateInterval`.

It's not a matter of code style, this test must succeed with a `DateInterval` instance and fail with a `CarbonInterval` or other classes that may extends `DateInterval`.

Comment (lucasmichot) [reply tests/CarbonInterval/ToDateIntervalTest.php:18]: > `is_a` has the same problem

I don't think so
```php
$this->assertInstanceOf(DateInterval::class, $interval);
$this->assertTrue(is_a($interval, DateInterval::class));
$this->assertFalse(is_a($interval, CarbonInterval::class));
```

Comment (lucasmichot) [reply tests/CarbonInterval/ToDateIntervalTest.php:18]: Nevertheless, I do not really care

Comment (lucasmichot) [reply tests/CarbonInterval/ToDateIntervalTest.php]: See c99a2c1bd8355c41f875abaeac0b5b32925241df

Comment (kylekatarnls) [reply tests/CarbonInterval/ToDateIntervalTest.php]: Again, `is_a(new CarbonInterval(), DateInterval::class)` returns true, which is wrong regarding the meaning of the test.

File: tests/CarbonPeriod/ToDatePeriodTest.php [hunk 2]
```diff
@@ -22,7 +23,7 @@ public function testToArrayIsNotEmptyArray()
         $result = $period->toDatePeriod();
 
         $this->assertFalse($period->isEndExcluded());
-        $this->assertSame('DatePeriod', get_class($result));
+        $this->assertInstanceOf(DatePeriod::class, $result);
```
Comment (kylekatarnls) [inline tests/CarbonPeriod/ToDatePeriodTest.php:26]: This test is still broken. `assertInstanceOf` is not what this test aims to verify.

File: tests/CarbonPeriod/ToDatePeriodTest.php [hunk 3]
```diff
@@ -35,7 +36,7 @@ public function testToArrayIsNotEmptyArray()
         $newInstance = CarbonPeriod::instance($result);
 
         $this->assertTrue($period->isEndExcluded());
-        $this->assertSame('DatePeriod', get_class($result));
+        $this->assertInstanceOf(DatePeriod::class, $result);
```
Comment (kylekatarnls) [inline tests/CarbonPeriod/ToDatePeriodTest.php:39]:
```suggestion
        $this->assertSame(DatePeriod::class, get_class($result));
```

File: tests/CarbonPeriod/ToDatePeriodTest.php [hunk 4]
```diff
@@ -47,7 +48,7 @@ public function testToArrayIsNotEmptyArray()
         $result = $period->toDatePeriod();
         $newInstance = CarbonPeriod::instance($result);
 
-        $this->assertSame('DatePeriod', get_class($result));
+        $this->assertInstanceOf(DatePeriod::class, $result);
```
Comment (kylekatarnls) [inline tests/CarbonPeriod/ToDatePeriodTest.php:51]:
```suggestion
        $this->assertSame(DatePeriod::class, get_class($result));
```

**Pull request closed without merging**